Document HTTP_PROXY and HTTPS_PROXY env vars for agent (#2856)
sensu/sensu-docs: content/sensu-go/6.1/observability-pipeline/observe-schedule/agent.md

For example, if you create a `SENSU_TEST_VAR` variable in your sensu-agent file, it will be available to use in your check configurations as `$SENSU_TEST_VAR`.

> ##### Use environment variables to define proxies for unconnected entities

Reviewer: ```suggestion
##### Use environment variables to specify an HTTP proxy for agent use
```

> To add dynamic runtime assets and run checks for entities that cannot access the internet, define `SENSU_HTTP_PROXY` and `SENSU_HTTPS_PROXY` environment variables in your sensu-agent file.

Reviewer: ```suggestion
If an HTTP proxy is required to access the internet in your compute environment, you may need to configure Sensu agent for this in order to successfully download dynamic runtime assets or execute commands which depend on internet access.

For Sensu agents which require a proxy server, define `HTTP_PROXY` and `HTTPS_PROXY` environment variables in your sensu-agent file.
```

> {{< code shell >}}
> SENSU_HTTP_PROXY="http://YOUR_PROXY_SERVER:PORT"
> SENSU_HTTPS_PROXY="http://YOUR_PROXY_SERVER:PORT"
> {{< /code >}}
> 
> {{% notice note %}}
> **NOTE**: You can use the same proxy server URL for `SENSU_HTTP_PROXY` and `SENSU_HTTPS_PROXY`.
> The proxy server URL you specify for `SENSU_HTTPS_PROXY` does not need to use `https://`.
> {{% /notice %}}
> 
> After you add the `SENSU_HTTP_PROXY` and `SENSU_HTTPS_PROXY` environment variables and restart sensu-agent, they will be available to check and hook commands executed by the Sensu agent.
> You can then use `SENSU_HTTP_PROXY` and `SENSU_HTTPS_PROXY` to add dynamic runtime assets, run checks, and complete other tasks that typically require an internet connection for your unconnected entities.

Reviewer: ```suggestion
{{< code shell >}}
HTTP_PROXY="http://YOUR_PROXY_SERVER:PORT"
HTTPS_PROXY="http://YOUR_PROXY_SERVER:PORT"
{{< /code >}}

{{% notice note %}}
**NOTE**: You can use the same proxy server URL for `HTTP_PROXY` and `HTTPS_PROXY`.
The proxy server URL you specify for `HTTPS_PROXY` does not need to use `https://`.
{{% /notice %}}

After you add the `HTTP_PROXY` and `HTTPS_PROXY` environment variables and restart sensu-agent, they will be available to check and hook commands executed by the Sensu agent.
You can then use `HTTP_PROXY` and `HTTPS_PROXY` to add dynamic runtime assets, run checks, and complete other tasks that typically require an internet connection for your unconnected entities.
```

As described in [this stackexchange post](https://unix.stackexchange.com/questions/212894/whats-the-right-format-for-the-http-proxy-environment-variable-caps-or-no-ca), an informal convention has emerged in Unixdom around `HTTP_PROXY` and `HTTPS_PROXY`, as well as the lower-case versions of these variables. 

As I understand, somewhere along the line our implementation uses the [ProxyFromEnvironment](https://golang.org/pkg/net/http/#ProxyFromEnvironment) function in Golang net/http to generate proxy settings based on the values of `HTTP_PROXY` and `HTTPS_PROXY`, but not the lower-case form of same.

Therefore, the environment variables shouldn't be prepended with `SENSU_` in this case.

Author: Thank you -- I wasn't sure about that but just followed what we had for other env vars.

For example, if you create a `SENSU_TEST_VAR` variable in your sensu-agent file, it will be available to use in your check configurations as `$SENSU_TEST_VAR`.

##### Use environment variables to specify an HTTP proxy for agent use

> If an HTTP proxy is required to access the internet in your compute environment, you may need to configure Sensu agent for this in order to successfully download dynamic runtime assets or execute commands which depend on internet access.

Author: ```suggestion
If an HTTP proxy is required to access the internet in your compute environment, you may need to configure the Sensu agent to successfully download dynamic runtime assets or execute commands that depend on internet access.
```

> For Sensu agents which require a proxy server, define `HTTP_PROXY` and `HTTPS_PROXY` environment variables in your sensu-agent file.

Author: ```suggestion
For Sensu agents that require a proxy server, define `HTTP_PROXY` and `HTTPS_PROXY` environment variables in your sensu-agent file.
```

{{< code shell >}}
HTTP_PROXY="http://YOUR_PROXY_SERVER:PORT"
HTTPS_PROXY="http://YOUR_PROXY_SERVER:PORT"
{{< /code >}}

> {{% notice note %}}
> **NOTE**: You can use the same proxy server URL for `HTTP_PROXY` and `HTTPS_PROXY`.
> The proxy server URL you specify for `HTTPS_PROXY` does not need to use `https://`.
> {{% /notice %}}

Author: ```suggestion
You can use the same proxy server URL for `HTTP_PROXY` and `HTTPS_PROXY`.
The proxy server URL you specify for `HTTPS_PROXY` does not need to use `https://`.
```

After you add the `HTTP_PROXY` and `HTTPS_PROXY` environment variables and restart sensu-agent, they will be available to check and hook commands executed by the Sensu agent.
You can then use `HTTP_PROXY` and `HTTPS_PROXY` to add dynamic runtime assets, run checks, and complete other tasks that typically require an internet connection for your unconnected entities.

> 

Author: ```suggestion

{{% notice note %}}
**NOTE**: If you define the `HTTP_PROXY` and `HTTPS_PROXY` environment variables, the agent WebSocket connection will also use the proxy URL you specify.
{{% /notice %}}
```
